nifi-nar-bundles/nifi-hubspot-bundle/nifi-hubspot-processors/src/main/java/org/apache/nifi/processors/hubspot/GetHubSpot.java

```diff
@@ -0,0 +1,320 @@
+                    " Customer Satisfaction (CSAT), Customer Effort Score (CES) and Custom Surveys."
+    );
+    static final AllowableValue LINE_ITEMS = new AllowableValue(
+            "/crm/v3/objects/line_items",
+            "Line Items",
+            "In HubSpot, line items can be thought of as a subset of products. When a product is attached to a deal, it becomes a line item. Line items can" +
+                    " be created that are unique to an individual quote, but they will not be added to the product library."
+    );
+    static final AllowableValue PRODUCTS = new AllowableValue(
+            "/crm/v3/objects/products",
+            "Products",
+            "In HubSpot, products represent the goods or services to be sold. Building a product library allows the user to quickly add products to deals," +
+                    " generate quotes, and report on product performance."
+    );
+    static final AllowableValue TICKETS = new AllowableValue(
+            "/crm/v3/objects/tickets",
+            "Tickets",
+            "In HubSpot, a ticket represents a customer request for help or support."
+    );
+    static final AllowableValue QUOTES = new AllowableValue(
+            "/crm/v3/objects/quotes",
+            "Quotes",
+            "In HubSpot, quotes are used to share pricing information with potential buyers."
+    );
+
+    // ENGAGEMENTS
+
+    private static final AllowableValue CALLS = new AllowableValue(
+            "/crm/v3/objects/calls",
+            "Calls",
+            "Get calls on CRM records and on the calls index page."
+    );
+    private static final AllowableValue EMAILS = new AllowableValue(
+            "/crm/v3/objects/emails",
+            "Emails",
+            "Get emails on CRM records."
+    );
+    private static final AllowableValue MEETINGS = new AllowableValue(
+            "/crm/v3/objects/meetings",
+            "Meetings",
+            "Get meetings on CRM records."
+    );
+    private static final AllowableValue NOTES = new AllowableValue(
+            "/crm/v3/objects/notes",
+            "Notes",
+            "Get notes on CRM records."
+    );
+    private static final AllowableValue TASKS = new AllowableValue(
+            "/crm/v3/objects/tasks",
+            "Tasks",
+            "Get tasks on CRM records."
+    );
+
+    // OTHER
+
+    private static final AllowableValue OWNERS = new AllowableValue(
+            "/crm/v3/owners/",
+            "Owners",
+            "HubSpot uses owners to assign specific users to contacts, companies, deals, tickets, or engagements. Any HubSpot user with access to contacts" +
+                    " can be assigned as an owner, and multiple owners can be assigned to an object by creating a custom property for this purpose."
+    );
+
+    static final PropertyDescriptor ACCESS_TOKEN = new PropertyDescriptor.Builder()
```

Comment:
The [HubSpot documentation](https://developers.hubspot.com/docs/api/working-with-oauth) indicates support for OAuth 2, as well as [Private access tokens](https://developers.hubspot.com/docs/api/intro-to-auth). At minimum, it would be helpful to have an `Authentication Strategy` property, which would provide the opportunity to configure with the access tokens, or OAuth 2.

Reply:
Thanks for the suggestion. Unfortunately they only support the Authorization Code grant type and our OAuthTokenProvider service works with Password and Client Credentials grant types.

Reply:
Thanks for the feedback @Lehel44, that is good to know. In that case, the current implementation makes sense. Just a question on the wording, should the property be named `Access Token` as opposed to `Admin API Access Token`?

Reply:
Thanks, it should be simply Access Token, I renamed it.